### PR TITLE
Initialization for the sinkhorn iterations

### PR DESCRIPTION
**Pull request by priba**
Following algorithm 2 of https://arxiv.org/pdf/1706.00292.pdf b is initialized to a vector of 1/m (however it can be weighted differently). Hence, with the change of variable, vector b should not be initialized as a vector of zeros.

Also, the default behaviour of nn is modified to ask the columns and rows of P sums up to one by default.

### CODE REVIEW

File: examples/sinkhorn_loss_functional/main.py [hunk 1]
```diff
@@ -47,11 +48,19 @@
     print(M)
 
     # Compute the transport matrix between each pair of sets in the minibatch with default parameters
-    P = sinkhorn(a, b, M, 1e-3)
-
+    P = sinkhorn(a, b, M, 1e-3, max_iters=5000, stop_thresh=1e-12)
```
Comment (fwilliams) [inline examples/sinkhorn_loss_functional/main.py]: You made `max_iters` 500 in the other example. Maybe keep this one 500 as well.

**Review by fwilliams — CHANGES_REQUESTED** [review]
Good catch on the non-uniform weight initialization! 

Any chance you could add an example for non-uniform weighting.

Other than that, I had a small nitpick which is a one character change, then I'll merge it.

File: examples/sinkhorn_loss_functional/main.py (hunk 1 above)
Comment (priba) [reply examples/sinkhorn_loss_functional/main.py]: It is already changed in the second commit.

**Comment by priba** [comment]
I just uploaded two examples and correct a bug:

1.- Weighted sets
2.- Unbalanced sets

The bug is the approximation error of P*1_d = a instead of P*b = a and vice-versa

**Pull request merged**